nixos/modules/security/systemd-confinement.nix

```diff
@@ -98,8 +98,9 @@ in {
         wantsAPIVFS = lib.mkDefault (config.confinement.mode == "full-apivfs");
       in lib.mkIf config.confinement.enable {
         serviceConfig = {
-          RootDirectory = "/var/empty";
-          TemporaryFileSystem = "/";
+          RuntimeDirectory = [ "confinement/${mkPathSafeName name}" ];
```

Comment:
I was wondering if any service use this for something already -- we might need to change this to modify the user's value e.g. confinement/name/run/oldvalue if there was one, so they'll have /run/oldvalue in the chroot
(note: after checking I have a couple of services with a RuntimeDirectory and confinement set (miniflux, postgresql), but nothing in there so they probably don't need it... I also have one that actually uses the directory (mpd), so I'll be able to report problem when testing in a bit)

Reply:
(still not tested) this is actually probably fine, I tested multiple RuntimeDirectory with a path within the chroot dir but using the option normally works as a list, so as long as we're working with services that properly define this as a list it'll work properly.

I don't think there's much we can do if a service defines it as a string though (would need a more complex merge mechanism with mixed types somewhere and I don't know how to do that in nix), we'll need to fix such services when coming accross them...

Reply:
Okay, that turned out quite painful as many of the services I use (postgres, mpd, miniflux, matrix-synapse...) define this as a string, and the merge policy fail when we have a list and a string: https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/systemd-unit-options.nix#L20

While I understand where that comes from and it's well intended I don't think it's actually that helpful - I guess if a user overrides some parameter with `serviceConfig.foo = bar` without override because the nixos service doesn't set it, and the nixos service starts adding one all of a sudden they'd be happy to get an error?...
So I'd be tempted to say we should just always concat+filter dups in there and wipe this under the rug, but that might not be satisfying for everyone.

I'm not sure, but as this I have just about as many overrides due to this than I had with dynamicUser = mkForce false, so we probably ought to come up with something here.